src/cluster/src/types.rs

```diff
@@ -0,0 +1,50 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! Types and traits required to setup clusters.
+
+use std::sync::Arc;
+
+use timely::scheduling::SyncActivator;
+use timely::worker::Worker as TimelyWorker;
+
+use mz_cluster_client::client::{ClusterStartupEpoch, TimelyConfig};
+use mz_persist_client::cache::PersistClientCache;
+type ActivatorSender = crossbeam_channel::Sender<SyncActivator>;
+
+pub trait AsRunningWorker<C, R> {
+    fn build_and_run<A: timely::communication::Allocate>(
+        config: Self,
+        timely_worker: &mut TimelyWorker<A>,
+        client_rx: crossbeam_channel::Receiver<(
+            crossbeam_channel::Receiver<C>,
+            tokio::sync::mpsc::UnboundedSender<R>,
+            ActivatorSender,
+        )>,
+        persist_clients: Arc<tokio::sync::Mutex<PersistClientCache>>,
+    );
+}
+
+pub trait CreatableCommand {
+    fn is_create_timely(self) -> Result<(TimelyConfig, ClusterStartupEpoch), Self>
```

Comment:
Not sure if this was considered but I would find a parent enum much easier to reason about:

```rust
enum ClusterCommand {
    CreateTimely(...),
    StorageCommand(StorageCommand),
    ComputeCommand(ComputeCommand),
}
```
Would something like this work? It will also allow us to move more common commands up in the future.

If we end up keeping this trait, I think we need a better name because `is_foo` methods are usually returning bool. `try_into_create_timely` would be one option, or somehow integrating with the `TryFrom` traits.

Reply:
A parent enum is a significant amount of code changes, that I am specifically avoiding until we merge the command streams!

the `try_into_` is great! ill probably keep the special trait but rename things